src/test/java/org/opensearch/flowframework/workflow/DeployModelStepTests.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.workflow;
+
+import com.carrotsearch.randomizedtesting.annotations.ThreadLeakScope;
+
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.ml.client.MachineLearningNodeClient;
+import org.opensearch.ml.common.MLTaskState;
+import org.opensearch.ml.common.MLTaskType;
+import org.opensearch.ml.common.transport.deploy.MLDeployModelResponse;
+import org.opensearch.test.OpenSearchTestCase;
+import org.opensearch.test.client.NoOpNodeClient;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import org.mockito.Answers;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doAnswer;
+
+@ThreadLeakScope(ThreadLeakScope.Scope.NONE)
+public class DeployModelStepTests extends OpenSearchTestCase {
+
+    private WorkflowData inputData = WorkflowData.EMPTY;
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    private NodeClient nodeClient;
+
+    @Mock
+    MachineLearningNodeClient machineLearningNodeClient;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+
+        inputData = new WorkflowData(Map.ofEntries(Map.entry("model_id", "modelId")));
+
+        MockitoAnnotations.openMocks(this);
+
+        nodeClient = new NoOpNodeClient("xyz");
+
+    }
+
+    public void testDeployModel() {
+
+        String taskId = "taskId";
+        String status = MLTaskState.CREATED.name();
+        MLTaskType mlTaskType = MLTaskType.DEPLOY_MODEL;
+
+        DeployModelStep deployModel = new DeployModelStep(nodeClient);
+
+        ArgumentCaptor<ActionListener<MLDeployModelResponse>> actionListenerCaptor = ArgumentCaptor.forClass(ActionListener.class);
+
+        doAnswer(invocation -> {
+            ActionListener<MLDeployModelResponse> actionListener = invocation.getArgument(2);
+            MLDeployModelResponse output = new MLDeployModelResponse(taskId, mlTaskType, status);
+            actionListener.onResponse(output);
+            return null;
+        }).when(machineLearningNodeClient).deploy(eq("modelId"), actionListenerCaptor.capture());
+
+        CompletableFuture<WorkflowData> future = deployModel.execute(List.of(inputData));
+
+        // TODO: Find a way to verify the below
+        // verify(machineLearningNodeClient).deploy(eq("modelId"), actionListenerCaptor.capture());
```

Comment:
If you change line 72 to :
`}).when(machineLearningNodeClient).deploy(any(), any());`

You can verify number of invocations like so : `verify(<mock>, times(<number of invocations>)).deploy(any(), actionListenerCapture.capture())`

And then assert the captured values
`assertEquals(<expected>, actionListenerCaptor.getValues().<MLDeployModelResponseGetterMethod()>)`

Reply:
Thanks for looking into it. Already tried the above approach and couple of other, it returns the below:
```java
  2> Wanted but not invoked:
    machineLearningNodeClient.deploy(
        "modelId",
        <Capturing argument: ActionListener>
    );
    -> at org.opensearch.flowframework.workflow.DeployModelStepTests.testDeployModel(DeployModelStepTests.java:79)
```

I have added the approaches tried [here](https://github.com/opensearch-project/opensearch-ai-flow-framework/pull/52#discussion_r1346548526). The issue is with mocking the [static](https://github.com/opensearch-project/opensearch-ai-flow-framework/blob/main/src/main/java/org/opensearch/flowframework/client/MLClient.java#L28) method.  It needs be mock like below:
```java
 try (MockedStatic<MLClient> mlClientMockedStatic = Mockito.mockStatic(MLClient.class)) {
      when(MLClient.getMLClient(any(NodeClient.class))).thenReturn(machineLearningNodeClient);
```
Which requires `mockito-inline` dependency and to avoid using that we have to use the below plugin `org.mockito.plugins.MockMaker ` with the extension `mock-maker-inline`. Tried both way which returns the below error
```
Caused by:
                org.mockito.exceptions.base.MockitoInitializationException: 
                Could not initialize inline Byte Buddy mock maker.
```
for which I added `net.bytebuddy` dependency  which again fails with the below

```
 Failed to load interface org.mockito.plugins.MockMaker implementation declared in java.lang.CompoundEnumeration@1f268988